testing/trino-testing/src/main/java/io/trino/testing/AbstractDynamicFilteringIntegrationSmokeTest.java

```diff
@@ -0,0 +1,759 @@
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, BROADCAST.name())
+                .setSystemProperty(JOIN_REORDERING_STRATEGY, NONE.name())
+                .build();
+    }
+
+    private Session withPartitionedJoin()
+    {
+        return Session.builder(this.getQueryRunner().getDefaultSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, PARTITIONED.name())
+                .build();
+    }
+
+    private static List<Integer> getOperatorRowsRead(DistributedQueryRunner runner, QueryId queryId)
+    {
+        QueryStats stats = runner.getCoordinator().getQueryManager().getFullQueryInfo(queryId).getQueryStats();
+        return stats.getOperatorSummaries()
+                .stream()
+                .filter(summary -> summary.getOperatorType().contains("Scan"))
+                .map(OperatorStats::getInputPositions)
+                .map(Math::toIntExact)
+                .collect(toImmutableList());
+    }
+
+    @Test
+    public void testJoinDynamicFilteringMultiJoin()
+    {
+        skipTestUnless(supportsNodeLocalDynamicFiltering());
+
+        assertUpdate("CREATE TABLE t0 (k0 integer, v0 real)");
+        assertUpdate("CREATE TABLE t1 (k1 integer, v1 real)");
+        assertUpdate("CREATE TABLE t2 (k2 integer, v2 real)");
+        assertUpdate("INSERT INTO t0 VALUES (1, 1.0)", 1);
+        assertUpdate("INSERT INTO t1 VALUES (1, 2.0)", 1);
+        assertUpdate("INSERT INTO t2 VALUES (1, 3.0)", 1);
+
+        String query = "SELECT k0, k1, k2 FROM t0, t1, t2 WHERE (k0 = k1) AND (k0 = k2) AND (v0 + v1 = v2)";
+        Session session = Session.builder(getSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, FeaturesConfig.JoinDistributionType.BROADCAST.name())
+                .setSystemProperty(JOIN_REORDERING_STRATEGY, FeaturesConfig.JoinReorderingStrategy.NONE.name())
+                .build();
+        assertQuery(session, query, "SELECT 1, 1, 1");
+    }
+
+    private void assertQueryResult(@Language("SQL") String sql, Object... expected)
+    {
+        MaterializedResult rows = computeActual(sql);
+        assertEquals(rows.getRowCount(), expected.length);
+
+        for (int i = 0; i < expected.length; i++) {
+            MaterializedRow materializedRow = rows.getMaterializedRows().get(i);
+            int fieldCount = materializedRow.getFieldCount();
+            assertEquals(fieldCount, 1, format("Expected only one column, but got '%d'", fieldCount));
+            Object value = materializedRow.getField(0);
+            assertEquals(value, expected[i]);
+            assertEquals(materializedRow.getFieldCount(), 1);
+        }
+    }
+
+    @Test(timeOut = 30_000)
+    public void testJoinWithEmptyBuildSide()
+    {
+        skipTestUnless(supportsCoordinatorDynamicFiltering());
+
+        ResultWithQueryId<MaterializedResult> result = getDistributedQueryRunner().executeWithQueryId(
+                getSession(),
```

Comment:
`getSession()` is too vague. it should be explicit from the name that joins are partitioned for coordinator tests

Reply:
Sorry @sopel39. Couldn't understand the concern here. Could you please explain bit further?

Reply:
Done